src/Raven.Server/Documents/Expiration/ExpirationStorage.cs

```diff
@@ -164,6 +164,7 @@ public record ExpiredDocumentsOptions
 
                                         expiredDocs.Add((clonedId, document.Id));
                                         totalCount++;
+                                        options.Context.Transaction.ForgetAbout(document);
```

Comment:
In the above call: `expiredDocs.Add((clonedId, document.Id));` we use `document.Id` property. It's `LazyStringValue` which points to a memory belonging to the decompression buffer.

Reply:
Okay, everything is fine because `expiredDocs` collection doesn't hold `LazyStringValue` but it has `string`. So during `expiredDocs.Add((clonedId, document.Id))` we materialized `LSV` to `string`.